kedro/framework/context/context.py

```diff
@@ -238,47 +240,61 @@ def _get_catalog(
             save_version=save_version,
         )
 
-        feed_dict = self._get_feed_dict()
-        catalog.add_feed_dict(feed_dict)
+        # TODO: [KDC] Need to confirm if it is better to have add_raw_data
+        # similar to add_feed_dict ?
+        raw_data = self._get_raw_data()
+
+        for ds_name, ds_data in raw_data.items():
+            dataset = (
```

Comment:
`catalog.__setter__` does these checks itself

```suggestion
  for ds_name, ds_data in raw_data.items():
      catalog[ds_name] = dataset
```